katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        # Restore original (slow) timestamps so that subsequent sensors (e.g. pointing) will have accurate values
+        extract_time = LazyTransform('extract_time', lambda t, keep: t + 0.5 * dump_period + time_offset)
+        self.sensor.timestamps = LazyIndexer(self._timestamps, keep=slice(num_dumps), transforms=[extract_time])
+        # Apply default selection and initialise all members that depend on selection in the process
+        self.select(spw=0, subarray=0, ants=script_ants)
+
+    @staticmethod
+    def _open(filename, mode='r'):
+        """Open file and do basic version and augmentation sanity check."""
+        f = h5py.File(filename, mode)
+        version = f.attrs.get('version', '1.x')
+        if not version == '2.5':
+            raise WrongVersion("Attempting to load version '%s' file with version 2.5 loader" % (version,))
+        if 'augment_ts' not in f.attrs:
+            raise BrokenFile('HDF5 file not augmented - please run dummy_augment.py in the scripts directory.')
+
+        return f, version
+
+    @staticmethod
+    def _get_ants(filename):
+        """Quick look function to get the list of antennas in a data file.
+
+        This is intended to be called without creating a full katdal object.
+
+        Parameters
+        ----------
+        filename : string
+            Data file name
+
+        Returns
+        -------
+        antennas : list of :class:'katpoint.Antenna' objects
+
+        """
+        f, version = H5DataV2_5._open(filename)
+        config_group = f['MetaData/Configuration']
+
+        # Only one antenna in an AVN file.
+        return [katpoint.Antenna(config_group["Antennas/ant1"].attrs["name"],
+                                 config_group["Antennas/ant1"].attrs["latitude"],
+                                 config_group["Antennas/ant1"].attrs["longitude"],
+                                 config_group["Antennas/ant1"].attrs["altitude"],
+                                 config_group["Antennas/ant1"].attrs["diameter"],
+                                 None,
+                                 make_pmodel_string(config_group["Antennas/ant1"]['pointing-model-params']),
+                                 config_group["Antennas/ant1"].attrs["beamwidth"]) ]
+
+    @staticmethod
+    def _get_targets(filename):
+        """Quick look function to get the list of targets in a data file.
+
+        This is intended to be called without createing a full katdal object.
+
+        Parameters
+        ----------
+        filename : string
+            Data file name
+
+        Returns
+        -------
+        targets : :class:'katpoint.Catalogue' object
+            All targets in file
+
+        """
+        f, version = H5DataV2_5._open(filename)
+        # Use the delay-tracking centre as the one and only target
+        # Try two different sensors for the DBE target
+        target_list = f['MetaData/Sensors/Antennas/ant1/target']
```

Comment:
This looks like the pointing centre, not the delay-tracking centre.

Reply:
Corrected in 9246136.